## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 426180`

Logs: https://github.com/Aleksanaa/nixpkgs-review-gha/actions/runs/16446964045


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.dmsuite</li>
    <li>python312Packages.dmsuite.dist</li>
    <li>python313Packages.dmsuite</li>
    <li>python313Packages.dmsuite.dist</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.dmsuite</li>
    <li>python312Packages.dmsuite.dist</li>
    <li>python313Packages.dmsuite</li>
    <li>python313Packages.dmsuite.dist</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.dmsuite</li>
    <li>python312Packages.dmsuite.dist</li>
    <li>python313Packages.dmsuite</li>
    <li>python313Packages.dmsuite.dist</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.dmsuite</li>
    <li>python312Packages.dmsuite.dist</li>
    <li>python313Packages.dmsuite</li>
    <li>python313Packages.dmsuite.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.dmsuite</summary>
<pre>        nodes = dmat.nodes
        assert np.allclose(nodes[0], dmat.xmin)
        assert np.allclose(nodes[-1], dmat.xmax)
        func = nodes**2
        dfunc = 2 * nodes
        d2func = 2.0
        d1_cheb = dmat.at_order(1) @ func
        d2_cheb = dmat.at_order(2) @ func
        d3_cheb = dmat.at_order(3) @ func
        assert np.allclose(d1_cheb, dfunc)
        assert np.allclose(d2_cheb, d2func)
>       assert np.allclose(d3_cheb, 0.0, atol=1e-6)
E       assert False
E        +  where False = <function allclose at 0x1062fabb0>(array([-1.27562089e-07, -3.15194484e-08,  3.28873284e-09,  8.14907253e-10,\n       -2.24190444e-10, -3.31965566e-11,  6...9,  4.74392436e-09,\n       -5.41331246e-09,  1.37370080e-08, -8.38190317e-09, -9.08970833e-07,\n       -1.10268593e-06]), 0.0, atol=1e-06)
E        +    where <function allclose at 0x1062fabb0> = np.allclose

tests/test_polydiff.py:49: AssertionError
=========================== short test summary info ============================
FAILED tests/test_polydiff.py::test_cheb_scaled - assert False
========================= 1 failed, 7 passed in 0.51s ==========================</pre>
</details>
<details>
<summary>python313Packages.dmsuite</summary>
<pre>        nodes = dmat.nodes
        assert np.allclose(nodes[0], dmat.xmin)
        assert np.allclose(nodes[-1], dmat.xmax)
        func = nodes**2
        dfunc = 2 * nodes
        d2func = 2.0
        d1_cheb = dmat.at_order(1) @ func
        d2_cheb = dmat.at_order(2) @ func
        d3_cheb = dmat.at_order(3) @ func
        assert np.allclose(d1_cheb, dfunc)
        assert np.allclose(d2_cheb, d2func)
>       assert np.allclose(d3_cheb, 0.0, atol=1e-6)
E       assert False
E        +  where False = <function allclose at 0x1028349f0>(array([-1.27562089e-07, -3.15194484e-08,  3.28873284e-09,  8.14907253e-10,\n       -2.24190444e-10, -3.31965566e-11,  6...9,  4.74392436e-09,\n       -5.41331246e-09,  1.37370080e-08, -8.38190317e-09, -9.08970833e-07,\n       -1.10268593e-06]), 0.0, atol=1e-06)
E        +    where <function allclose at 0x1028349f0> = np.allclose

tests/test_polydiff.py:49: AssertionError
=========================== short test summary info ============================
FAILED tests/test_polydiff.py::test_cheb_scaled - assert False
========================= 1 failed, 7 passed in 0.51s ==========================</pre>
</details>
</details>

--- a/pkgs/development/python-modules/dmsuite/default.nix
+++ b/pkgs/development/python-modules/dmsuite/default.nix
@@ -1,0 +1,54 @@
+{
+  lib,
+  stdenv,
+  buildPythonPackage,
+  fetchPypi,
+  numpy,
+  pythonOlder,
+  pytestCheckHook,
+  setuptools,
+  setuptools-scm,
+  scipy,
+}:
+
+buildPythonPackage rec {
+  pname = "dmsuite";
+  version = "0.3.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-IqLsHkGNgPz2yZm0QMyMMo6Mr2RsU2DPGxYpoNwC3fs=";
+  };
+
+  build-system = [
+    setuptools
+    setuptools-scm
+  ];
+
+  dependencies = [
+    numpy
+    scipy
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  # Slight precision error probably due to different BLAS backend on Darwin
+  disabledTests = lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) [
+    "test_cheb_scaled"
+  ];
+
+  pythonImportsCheck = [ "dmsuite" ];
+
+  meta = {
+    description = "Scientific library providing a collection of spectral collocation differentiation matrices";
+    homepage = "https://github.com/labrosse/dmsuite";
+    changelog = "https://github.com/labrosse/dmsuite/releases/tag/v${version}";
+    license = lib.licenses.gpl2Only;
+    maintainers = with lib.maintainers; [ loicreynier ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -4152,6 +4152,8 @@ self: super: with self; {
 
   dmgbuild = callPackage ../development/python-modules/dmgbuild { };
 
+  dmsuite = callPackage ../development/python-modules/dmsuite { };
+
   dmt-core = callPackage ../development/python-modules/dmt-core { };
 
   dnachisel = callPackage ../development/python-modules/dnachisel { };

